src/model_repository_manager/model_lifecycle.cc

```diff
@@ -799,10 +801,11 @@ ModelLifeCycle::OnLoadFinal(
     // Mark current versions ready and track info in foreground
     for (auto& loaded : load_tracker->load_set_) {
       std::lock_guard<std::mutex> curr_info_lk(loaded.second->mtx_);
-
       loaded.second->state_ = ModelReadyState::READY;
       loaded.second->state_reason_.clear();
-
+#ifdef TRITON_ENABLE_METRICS
+      CalculateAndReportLoadTime(loaded.second);
+#endif  // TRITON_ENABLE_METRICS
```

Comment:
This metric is per model so 0.2 and 10 seconds would be reported and stored for each of the model
https://github.com/triton-inference-server/core/pull/397/files#diff-d459d645388f914acbe22d1d7b9b689e2a1a349b2ab899df4c88de66c21235a0R474
We set the load_start_ns_ per model ONLY when it starts LOADING and calculated when model is READY
Below is example with multiple models loaded at startup

`# HELP nv_model_load_time Load Time per-model in seconds
# TYPE nv_model_load_time gauge
nv_model_load_time{model="simple_string",version="1"} 0.790762448
nv_model_load_time{model="simple_sequence",version="1"} 0.7909725990000001
nv_model_load_time{model="densenet_onnx",version="1"} 0.964002791
nv_model_load_time{model="simple_int8",version="1"} 0.775782746
nv_model_load_time{model="simple_identity",version="1"} 0.728239961
nv_model_load_time{model="simple",version="1"} 0.7150264079999999
nv_model_load_time{model="simple_dyna_sequence",version="1"} 0.730162601
nv_model_load_time{model="inception_graphdef",version="1"} 0.929637874`

Did I miss something?

Reply:
My bad, I mean across versions of the same model.

Reply:
```
model_repo
|
`-- model_dir
    |
    `-- 1
    |   |
    |   `-- model.py   <-- takes 0.2 seconds to load
    |
    `-- 2
    |   |
    |   `-- model.py  <-- takes 10 seconds to load
    |
    `-- config.pbtxt
```
If both versions are loaded at the same time, is the metrics going to show
```
nv_model_load_time{model="model_name",version="1"} 0.2
nv_model_load_time{model="model_name",version="2"} 10
```

Reply:
Yes, because we load versions sequentially
OnLoadComplete is where we set the metric
https://github.com/triton-inference-server/core/blob/571c5dcee325a78cd8f02e2ee8ebb17e1df2ac75/src/model_repository_manager/model_lifecycle.cc#L505
load_start_ns_ is set again for every version
https://github.com/triton-inference-server/core/blob/571c5dcee325a78cd8f02e2ee8ebb17e1df2ac75/src/model_repository_manager/model_lifecycle.cc#L467
We set load_start_ns_ in this loop ensuring this case.
Will post a local test result here

Reply:
Below is the output of the config you explained

`nv_model_load_time{model="simple_string",version="1"} 0.282414657
nv_model_load_time{model="simple_sequence",version="1"} 0.28993499
nv_model_load_time{model="simple_identity",version="1"} 0.262991678
nv_model_load_time{model="densenet_onnx",version="1"} 0.352527317
nv_model_load_time{model="simple",version="1"} 0.216015252
nv_model_load_time{model="inception_graphdef",version="1"} 0.362119909
nv_model_load_time{model="simple_dyna_sequence",version="1"} 0.233802043
nv_model_load_time{model="simple_int8",version="1"} 0.265711744
nv_model_load_time{model="simple",version="2"} 0.216017807`

You can see different load times for `simple` model with different versions

Reply:
What happen if you add a 10 seconds delay to 
```
nv_model_load_time{model="simple",version="1"} 0.216015252
```
model version? Does it not affect
```
nv_model_load_time{model="simple",version="2"} 0.216017807
```
?

Reply:
>  we load versions sequentially

FYI I believe the versions can be loaded in parallel: https://github.com/triton-inference-server/core/blob/c59250656e28636db9df079bf4693e382bda3c55/src/model_repository_manager/model_lifecycle.cc#L532-L534

Reply:
Yes, I tested this particular scenario turns out due to async load subsequent versions of the model have in-correct timing even though in reality the time shoud be reflected only in version 1.

`nv_model_load_time{model="simple",version="1"} 10.216015252
nv_model_load_time{model="simple",version="2"} 10.216017807`

Syncing up with @kthui in sometime for this

Reply:
Synced offline, if the `nv_model_load_time` metrics only care about the the duration for loading a new model (version), then we can simply time the duration for executing [this block](https://github.com/triton-inference-server/core/blob/main/src/backend_model.cc#L68-L232) of `TritonModel::Create()` code. We could have a separate metrics (i.e. `nv_model_update_time`) for tracking the update duration of a loaded model (version) by timing [this block](https://github.com/triton-inference-server/core/blob/main/src/backend_model.cc#L239-L270) of `TritonModel::UpdateInstanceGroup()` code.

Reply:
Fixed and tested as per our discussion.
` HELP nv_model_load_duration_secs Model load time in seconds
# TYPE nv_model_load_duration_secs gauge
nv_model_load_duration_secs{model="input_all_optional",version="2"} 1.529608532
nv_model_load_duration_secs{model="input_all_optional",version="1"} 11.699666445`

Reply:
Nice job fixing the logic issue!